examples/platform/silabs/efr32/rs911x/hal/sl_board_configuration.h

```diff
@@ -0,0 +1,57 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#ifndef _RSI_BOARD_CONFIGURATION_H_
+#define _RSI_BOARD_CONFIGURATION_H_
+
+typedef struct
+{
+    unsigned char port;
+    unsigned char pin;
+} sl_pin_t;
+
+#define PIN(port_id, pin_id)                 \
+  (sl_pin_t)                                 \
+  {                                          \
+    .port = gpioPort##port_id, .pin = pin_id \
+  }
+
+#define PACKET_PENDING_INT_PRI 3
+
+#if defined(EFR32MG12_BRD4161A) || defined(BRD4161A) || defined(EFR32MG12_BRD4162A) || defined(BRD4162A) ||                        \
+    defined(EFR32MG12_BRD4163A) || defined(BRD4163A) || defined(EFR32MG12_BRD4164A) || defined(BRD4164A) ||                        \
+    defined(EFR32MG12_BRD4170A) || defined(BRD4170A)
+// BRD4161-63-64 are pin to pin compatible for SPI
+#include "brd4161a.h"

```

Comment:
Those header file should be renamed to a common generic name that each board has. Instead of a bunch of ifdef include. You have only one include of this generic named file